src/puppetlabs/dujour/version_check.clj

```diff
@@ -88,19 +95,37 @@
                          (assoc "group" group-id)
                          (merge version-data)
                          json/generate-string)
-        url update-server-url
+        _ (log/tracef "Making update request to %s with data: %s" update-server-url request-body)
         {:keys [status body] :as resp} (try
-                                         (client/post url
-                                                     {:headers {"Accept" "application/json"}
-                                                      :body request-body
-                                                      :as :text})
+                                         (client/post update-server-url
+                                                      {:headers {"Accept" "application/json"}
+                                                       :body request-body
+                                                       :as :text})
+                                         (catch HttpException e
+                                           (throw-connection-error e))
                                          (catch IOException e
-                                           (throw+ {:kind ::connection-error
-                                                    :msg (.getMessage e)
-                                                    :cause e})))]
-      {:status status :body body :resp resp}))
-
-(schema/defn ^:always-validate update-info :- (schema/maybe UpdateInfo)
+                                           (throw-connection-error e)))]
+
+    (log/tracef "Received response from %s, status: %s, body: %s" update-server-url status body)
+    {:status status :body body :resp resp}))
+
+(defn- throw-unexpected-response
+  [body]
+  (throw+ {:kind ::unexpected-response
+           :msg "Server returned HTTP code 200 but the body was not understood."
+           :details {:body body}}))
```

Comment:
Would it be worth moving this and `throw-connection-error` into their own namespace?